### ISSUE TITLE
finally find property super parameters for mini-imagenet

### ISSUE THREAD

**Issue by KaleidoZhouYN** [wyharveychen/CloserLookFewShot#12]
after a lot of attemptions,i finally found a robust parameters to re-implement the refined cosine distance method:
1.we rewrite the cosine distance code.
2. we set scale factor s = 30 all the time for training.
3. we use sgd optimizer with m=0.9 and lr=0.1 at begining,and decay with 0.1 at 100,200,300 epoch
4.for conv4 network we use weight-decay = 5e-4,for ResNetX network we use weight-decay = 1e-3
5.when doing finetune,we set the scale factor s = 5.

here is the result on ResNet18:
mini-imagenet:
     1 shot 56.78+0.84  
     5 shot 75.46+0.63

**Comment by wyharveychen** [comment]
Oh my god, thanks for sharing, and this is really helpful!
Looks like there are still many things for me to learn how to make the learning more stable...
Do you mind to upload your code as a new branch? It would be great to have this version available!

**Comment by a1600012888** [comment]
> after a lot of attemptions,i finally found a robust parameters to re-implement the refined cosine distance method:
> 1.we rewrite the cosine distance code.
> 2. we set scale factor s = 30 all the time for training.
> 3. we use sgd optimizer with m=0.9 and lr=0.1 at begining,and decay with 0.1 at 100,200,300 epoch
> 4.for conv4 network we use weight-decay = 5e-4,for ResNetX network we use weight-decay = 1e-3
> 5.when doing finetune,we set the scale factor s = 5.
> 
> here is the result on ResNet18:
> mini-imagenet:
> 1 shot 56.78+0.84
> 5 shot 75.46+0.63

Thanks for these hyper-params. Can you still remember the hyper-param when we do fine-tuning for baseline++?

**Comment by tao0420** [comment]
@KaleidoZhouYN Thanks for providing this info! I am wondering what is the final training loss value(for l2-normalized softmax) you get under this setting of hyperparameter?

**Comment by banjuanshua** [comment]
Can you explain how do you find these super parameters?

**Comment by weitliu** [comment]
I just feel like if you end up with a very different optimizer and learning rate scheduler, the performance of other benchmarks may improve or worse as well. The values reported may have to be updated as well...
